### PR TITLE
Remove version agnostic operator name mapping

### PR DESCRIPTION
**Pull request by vshampor**
The PyTorch public API is quite stable now, so there is no need to do this bookkeeping. This will, however, impact the majority of the existing .pth checkpoints as they would have to be converted so that the activation quantizer identifiers have the version agnostic `RELU` replaced with either `relu` or `relu_`.

### CODE REVIEW

**Comment by vshampor** [comment]
Pytorch eval build 381 OK with existing checkpoints

File: nncf/torch/checkpoint_loading.py [hunk 1]
```diff
@@ -268,8 +268,26 @@ def run(self) -> Dict[str, torch.Tensor]:
...
+        if has_version_agnostic_names:
+            warnings.warn('Legacy NNCF-enabled .pth checkpoint has been loaded! '
+                          'The version-agnostic `RELU` operator name entries in the state dict have been deprecated. '
+                          'The loader will try to match these entries to the correspoindig `relu` and `relu_` op '
+                          'names. The newly exported checkpoints will be adjusted to the new format.',
+                          category=DeprecationWarning)
+
         for normalized_key_to_load in normalized_keys_to_load:
             key_to_load = normalized_keys_to_load.get_orig_key(normalized_key_to_load)
+            normalized_key_to_load = cross_match_key_map.get(normalized_key_to_load, normalized_key_to_load)
```
Comment (ljaljushkin) [inline nncf/torch/checkpoint_loading.py:289]: This function call looks strange. should arguments really be duplicated ?

File: nncf/torch/checkpoint_loading.py [hunk 2]
```diff
@@ -268,8 +268,26 @@ def run(self) -> Dict[str, torch.Tensor]:
         normalized_keys_to_load = NormalizedKeys(list(self.state_dict_to_load.keys()))
 
         has_legacy_storage_keys = False
+
+        has_version_agnostic_names = False
```
Comment (ljaljushkin) [inline nncf/torch/checkpoint_loading.py:272]: please move this functionality to `NormalizedKeys` class as it was done in my PR for legacy_storage_keys: 
https://github.com/openvinotoolkit/nncf/pull/682/files#diff-40c3a1a297ae11ac63166f5c9a2117bb25bd2e0c548e6d36e07edf64bd9848dfR241

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: nncf/torch/checkpoint_loading.py (hunk 1 above)
Comment (vshampor) [reply nncf/torch/checkpoint_loading.py:289]: I tried to do the same using an `in` check, and Pylint triggered on it, advising to use `get`. `dict.get` returns the second argument value if the first argument value was not found among the keys of the dict.

File: nncf/torch/checkpoint_loading.py (hunk 2 above)
Comment (vshampor) [reply nncf/torch/checkpoint_loading.py:272]: The matching done in this PR cannot be done in `NormalizedKeys` because it requires both the model and the checkpoint key sets - we don't know whether `RELU` in a given model corresponds to `relu` or `relu_`, so we have to iterate over alternatives.

**Review by ljaljushkin — APPROVED** [review]

**Comment by vshampor** [comment]
Jenkins please retry a build
